src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,69 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React, { Dispatch, SetStateAction } from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+interface T {
+  setFilterType: Dispatch<SetStateAction<string>>;
+  input: string;
+  setInput: Dispatch<SetStateAction<string>>;
+}
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<T> = ({
+  input,
+  setFilterType,
+  setInput,
+}) => {
+  const handleFilterChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setFilterType(event.target.value);
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select data-cy="statusSelect" onChange={handleFilterChange}>
+            <option
+              value="all"
+            >
+              All
+            </option>
+            <option
+              value="active"
+            >
+              Active
+            </option>
+            <option
+              value="completed"
+            >
+              Completed
+            </option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={input}
+          onChange={(event) => setInput(event.target.value)}
```

Comment:
```suggestion
          onChange={your_function_name}
```

Consider moving this logic from jsx